### PR TITLE
IMDS Swagger: Adding Swagger spec for 2019-08-15 version

### PR DESCRIPTION
**Pull request by jianyunt**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x]  I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x]  [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by NullMDR** [comment]
@openapibot sdkautomation rebuild

**Comment by jianyunt** [comment]
Hi @lirenhe and other admins, all checks are passed now. Please review the changes and approve and merge if possible. Thanks!

\\cc @rifrankl @rikotcho @KumariSupriya @harijayms @edyoung

**Comment by jianyunt** [comment]
Hi @lirenhe and other admins, I have integrated the changes from IMDS 2019-08-01 and all checks are passed.  Please let us know if anything we need to do to get the PR approved and merged?

@rifrankl @rikotcho @KumariSupriya @harijayms @edyoung

**Review by ryansbenson — COMMENTED** [review]
Data plane APIs aren't reviewed by ARM. Removing label.

**Review by lirenhe — APPROVED** [review]

File: specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2019-08-15/imds.json
Comment (johanste) [inline specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2019-08-15/imds.json]: Changing the name of the parameter is a breaking change (it was `msi_res_id` in the previous version). Why is this change necessary?

File: specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2019-08-15/imds.json
Comment (johanste) [inline specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2019-08-15/imds.json]: From an API perspective, I have no concerns. However, if this document is to be used for autorest, adding additional failure cases in the responses section will negatively impact the generated code.

**Review by PhoenixHe-NV — COMMENTED** [review]
Why we only have python sdk to generate in swagger-to-sdk section? Do you want to add other sdks?

File: specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2019-08-15/imds.json
Comment (jianyunt) [reply specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2019-08-15/imds.json]: The name change was based on code review feedback from the Identity team. But I leave it out for now I changed back to `msi_res_id`.

Comment (johanste) [reply specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2019-08-15/imds.json]: What does the API actually expect? Was the description incorrect earlier and this was a correction, or was this a planned change to the REST API?

Comment (jianyunt) [reply specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2019-08-15/imds.json]: I am not 100% sure if imds is responding mi_res_id today.  So I'd rather leave this particular change to the Identity team.

**Comment by jianyunt** [comment]
@NullMDR I am not sure if I got your point. And I see the build seems to cover sdk node, sdk java, go, python, etc. already.

File: specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2019-08-15/imds.json
Comment (jianyunt) [reply specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2019-08-15/imds.json]: I removed 401 and added headers to default section. As headers is optional, so we won't break the existing scenarios. Thanks @johanste.

File: specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2019-08-15/imds.json
Comment (johanste) [reply specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2019-08-15/imds.json]: This needs to be sorted out before we can merge the PR. The description has to be correct w. regards to the actual REST API.

**Comment by NullMDR** [comment]
@jianyunt I mean the section here https://github.com/Azure/azure-rest-api-specs/pull/6846/files#diff-1f6ca4d24ed36009a22cccc573434122R223

We will move to new automation system in a near furture and you need that section to contain configuration for other languages to generate sdk in the new system.

**Comment by jianyunt** [comment]
@NullMDR what specific file do I need to change and how in order to adopt the new system you mentioned to cover other languages? it should be happening now or later?

File: specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2019-08-15/imds.json
Comment (jianyunt) [reply specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2019-08-15/imds.json]: @johanste , do you mean whether `msi_res_id` or `mi_res_id` should be sorted it now? I looked file history, it has been `msi_res_id`. So no change for this version.

Comment (jianyunt) [reply specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2019-08-15/imds.json]: @johanste I looked identity team's source code, there are no changes in the history about msi_res_id. Also public docs are about msi_res_id. 
This version is particularly for our hybrid scenario. No impact to the existing Azure IMDS. 
If you have concerns, I can follow-up with the Metadata and Identity teams about it. However to fix the existing spec issues, can we  make a separate PR to fix the existing api versions all together. Thoughts?

Comment (johanste) [reply specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2019-08-15/imds.json]: Assuming that the name in the swagger is what the REST API actually expects/uses, then we are good. From you explanation, that seems to be the case. What we should *not* do is check in a swagger that is incorrect and then later fix it. 

If the service team intends to change the name in the REST API in the future, please make sure that they consult the azure api review board to determine if the change is warranted.

**Comment by johanste** [comment]
No actual API changes. No need for the Azure API review board to review.

File: specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2019-08-15/imds.json
Comment (jianyunt) [reply specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2019-08-15/imds.json]: Agreed.  teams should consult with the board first for any breaking changes. I will follow-up with the Metadata and Identity team to see if there are any inconsistencies between code and the spec.
For this particular PR, do you have any other concerns?

**Comment by NullMDR** [comment]
@jianyunt https://github.com/Azure/azure-rest-api-specs/blob/f0938f745e66b378b3874224a929a56ff88cf5e9/specification/imds/data-plane/readme.md#swagger-to-sdk 

You'd better update it before we switch to new sdk generation system. We'll upgrade to new system a few month later.

**Pull request merged**
